skymp5-server/cpp/server_guest_lib/MpObjectReference.cpp

```diff
@@ -1748,3 +1767,69 @@ void MpObjectReference::BeforeDestroy()
 
   RemoveFromGridAndUnsubscribeAll();
 }
+
+bool MpObjectReference::MpApiOnPutItem(MpActor& source,
+                                       const Inventory::Entry& entry)
+{
+  simdjson::dom::parser parser;
+  std::string rawArgs = "[" + std::to_string(source.GetFormId()) + "," +
+    std::to_string(entry.baseId) + "," + std::to_string(entry.count) + "]";
+  auto args = parser.parse(rawArgs).value();
+  bool blockedByMpApi = false;
+
+  if (auto wst = GetParent()) {
+    const auto id = GetFormId();
+    for (auto& listener : wst->listeners) {
+      bool notBlocked = listener->OnMpApiEvent("onPutItem", args, id);
+      blockedByMpApi = !notBlocked;
+    }
+  }
+
+  return blockedByMpApi;
+}
+
+bool MpObjectReference::MpApiOnTakeItem(MpActor& source,
+                                        const Inventory::Entry& entry)
+{
+  simdjson::dom::parser parser;
+  std::string rawArgs = "[" + std::to_string(source.GetFormId()) + "," +
+    std::to_string(entry.baseId) + "," + std::to_string(entry.count) + "]";
+  auto args = parser.parse(rawArgs).value();
+  bool blockedByMpApi = false;
+
+  if (auto wst = GetParent()) {
+    const auto id = GetFormId();
+    for (auto& listener : wst->listeners) {
+      bool notBlocked = listener->OnMpApiEvent("onTakeItem", args, id);
+      blockedByMpApi = !notBlocked;
+    }
+  }
+
+  return blockedByMpApi;
+}
+
+float MpObjectReference::GetTotalItemWeight() const
+{
+  const auto& entries = GetInventory().entries;
+  const auto calculateWeight = [this](float sum,
+                                      const Inventory::Entry& entry) {
+    const auto& espm = GetParent()->GetEspm();
+    const auto* record = espm.GetBrowser().LookupById(entry.baseId).rec;
+    if (!record) {
+      spdlog::trace("Record of form ({}) is nullptr", entry.baseId);
```

Comment:
better mention function in logs, otherwise will be hard to find 

like "MpObjectReference::GetTotalItemWeight :x - ... " where :x is this->GetFormId()

also, log level should be at least warn I guess.